// angular2-useful-swiper/fesm2015/angular2-useful-swiper.js
import { CommonModule } from '@angular/common';
import { Component, ElementRef, ChangeDetectorRef, Input, NgModule } from '@angular/core';
import Swiper from 'swiper';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class SwiperComponent {
    /**
     * @param {?} elementRef
     * @param {?} changeDetectorRef
     */
    constructor(elementRef, changeDetectorRef) {
        this.elementRef = elementRef;
        this.changeDetectorRef = changeDetectorRef;
        this.slideCount = 0;
        this.initialized = false;
        this.shouldInitialize = true;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set initialize(value) {
        this.shouldInitialize = this.initialized ? false : value;
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        if (this.shouldInitialize) {
            this.setup();
        }
    }
    /**
     * @return {?}
     */
    setup() {
        if (!this.swiper) {
            // if rendered on server querySelector is undefined
            if (this.elementRef.nativeElement.querySelector) {
                this.swiperWrapper = this.elementRef.nativeElement.querySelector('.swiper-wrapper');
                this.slideCount = this.swiperWrapper.childElementCount;
                this.swiper = new Swiper(this.elementRef.nativeElement.querySelector('swiper > div'), this.config);
                this.changeDetectorRef.detectChanges();
            }
            this.shouldInitialize = false;
        }
    }
    /**
     * @return {?}
     */
    ngAfterViewChecked() {
        if (this.shouldInitialize) {
            this.setup();
        }
        if (this.swiperWrapper &&
            this.slideCount !== this.swiperWrapper.childElementCount) {
            this.slideCount = this.swiperWrapper.childElementCount;
            this.swiper.update();
        }
    }
}
SwiperComponent.decorators = [
    { type: Component, args: [{
                // tslint:disable-next-line: component-selector
                selector: 'swiper',
                template: "<div\n  [ngClass]=\"{\n    'swiper-container': config?.containerModifierClass === undefined\n  }\"\n>\n  <ng-content></ng-content>\n</div>\n",
                styles: [":host{display:block}.swiper-container{width:100%;height:100%}"]
            }] }
];
/** @nocollapse */
SwiperComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: ChangeDetectorRef }
];
SwiperComponent.propDecorators = {
    config: [{ type: Input }],
    initialize: [{ type: Input, args: ['initialize',] }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class Angular2UsefulSwiperModule {
}
Angular2UsefulSwiperModule.decorators = [
    { type: NgModule, args: [{
                declarations: [SwiperComponent],
                imports: [CommonModule],
                exports: [SwiperComponent]
            },] }
];

export { Angular2UsefulSwiperModule, SwiperComponent };
